File: Dot examples/ASP Book Examples/Book Examples/Chapter29/VB/App_Code/MovieView.vb
Imports System
Imports System.Web.UI
Imports System.Web.UI.WebControls
Imports System.ComponentModel

Namespace myControls

    <DefaultProperty("Title")> _
    Public Class MovieView
        Inherits WebControl

        Private _title As String = "Movie Title"
        Private _description As String = "Movie Description"

        <Category("Movie")> _
        <Description("Movie Title")> _
        Public Property Title() As String
            Get
                Return _title
            End Get
            Set(ByVal Value As String)
                _title = Value
            End Set
        End Property

        <Category("Movie")> _
        <Description("Movie Description")> _
        Public Property Description() As String
            Get
                Return _description
            End Get
            Set(ByVal Value As String)
                _description = Value
            End Set
        End Property


        Protected Overrides Sub RenderContents(ByVal writer As HtmlTextWriter)
            writer.RenderBeginTag(HtmlTextWriterTag.H1)
            writer.Write(_title)
            writer.RenderEndTag()

            writer.Write(_description)
        End Sub

        Protected Overrides ReadOnly Property TagKey() As HtmlTextWriterTag
            Get
                Return HtmlTextWriterTag.Div
            End Get
        End Property
    End Class
End Namespace

File: Dot examples/ASP Book Examples/Book Examples/Chapter29/CS/App_Code/MovieView.cs
using System;
using System.Web.UI;
using System.Web.UI.WebControls;
using System.ComponentModel;

namespace myControls
{
    [DefaultProperty("Title")]
    public class MovieView : WebControl
    {

        private string _title = "Movie Title";
        private string _description = "Movie Description";

        [Category("Movie")]
        [Description("Movie Title")]
        public string Title
        {
            get { return _title; }
            set { _title = value; }
        }

        [Category("Movie")]
        [Description("Movie Description")]
        public string Description
        {
            get { return _description; }
            set { _description = value; }
        }


        protected override void RenderContents(HtmlTextWriter writer)
        {
            writer.RenderBeginTag(HtmlTextWriterTag.H1);
            writer.Write(_title);
            writer.RenderEndTag();

            writer.Write(_description);
        }

        protected override HtmlTextWriterTag TagKey
        {
            get
            {
                return HtmlTextWriterTag.Div;
            }
        }
    }
}